src/Surfnet/ServiceProviderDashboard/Application/Console/CleanOldProductionEntitiesCommand.php

```diff
@@ -0,0 +1,104 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\ServiceProviderDashboard\Application\Console;
+
+use Surfnet\ServiceProviderDashboard\Domain\Entity\Entity;
+use Surfnet\ServiceProviderDashboard\Domain\Repository\EntityRepository;
+use Surfnet\ServiceProviderDashboard\Infrastructure\Manage\Client\QueryClient;
+use Surfnet\ServiceProviderDashboard\Infrastructure\Manage\Exception\QueryServiceProviderException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class CleanOldProductionEntitiesCommand extends Command
+{
+    /**
+     * @var EntityRepository
+     */
+    private $entityRepository;
+    /**
+     * @var QueryClient
+     */
+    private $productionManageClient;
+
+    /**
+     * CleanOldProductionEntitiesCommand constructor.
+     * @param EntityRepository $entityRepository
+     * @param QueryClient $manageClient
+     */
+    public function __construct(EntityRepository $entityRepository, QueryClient $productionManageClient)
+    {
+        $this->entityRepository = $entityRepository;
+        $this->productionManageClient = $productionManageClient;
+
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('app:clear-old-entities')
+            ->setDescription('Clears manage production entities not known to Manage.')
+            ->setHelp('This command allows you to clear old published production entities not known in to the production Manage.');
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return int|null|void
+     * @throws QueryServiceProviderException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $output->write('Start');
+        do {
+            $output->write('.');
+        } while ($this->updateNextBatch());
+        $output->write("\nDone\n");
+    }
+
+    /**
+     * @return bool
+     * @throws QueryServiceProviderException
+     */
+    private function updateNextBatch()
+    {
+        $clearEntities = [];
+        
+        // fetch published production entities
+        $entities = $this->entityRepository->findByState(Entity::STATE_PUBLISHED, Entity::ENVIRONMENT_PRODUCTION);
+        if (empty($entities)) {
+            return false;
+        }
+
+        // check if entities are in manage and if not add to queue
+        foreach ($entities as $entity) {
+            $result = $this->productionManageClient->findByManageId($entity->getManageId());
```

Comment:
Whats the point in checking if the entity exists in Manage?

If it does: *It would have been shown as a duplicate and we want to delete the local entity*
If it doesn't: *The entity was never published by the service desk, possibly mail service desk, but still remove the entity from local storage* 

There is another possibility; that a prod entity was requested for publishing in manage (the coin:exclude-from-push attr was set). In that case, we still want to remove the corresponding 'local prod published' entity right?

Please verify this assumption with @quartje and @teunfransen.

Reply:
I added this check because If someone later on accidentally runs this console action I don't want to remove the local production entities.

Reply:
After verifying this with @quartje the check in manage doesn't have to exist because we could rely on the fact that published to production entities should never exists in sp-dashboard and should only exist in manage.

Reply:
Therefore I removed this check.

Reply:
Thanks! The help message could be updated accordingly.